session/txn.go

```
@@ -157,6 +158,15 @@ func (st *TxnState) Commit(ctx context.Context) error {
 		st.fail = nil
 		return errors.Trace(err)
 	}
+	if len(st.mutations) != 0 || len(st.dirtyTableOP) != 0 || st.buf.Len() != 0 {
+		log.Errorf("The code should never run here, TxnState=%#v, mutations=%#v, dirtyTableOP=%#v, buf=%#v something must be wrong: %s",
+			st,
+			st.mutations,
+			st.dirtyTableOP,
+			st.buf,
+			debug.Stack())
+		st.cleanup()
```

Should we return an error here？

---

done